src/casync-http.c

```diff
@@ -32,6 +33,39 @@ typedef enum ProcessUntil {
         PROCESS_UNTIL_FINISHED,
 } ProcessUntil;
 
+static CURLcode robust_curl_easy_perform(CURL *curl) {
+        useconds_t sleep_time = 100000;
+        uint trial = 1;
+        uint limit = 10;
+        CURLcode c;
+
+        assert(curl);
+
+        while (trial < limit) {
+
+                c = curl_easy_perform(curl);
+
+                switch (c) {
+
+                case CURLE_COULDNT_CONNECT:
+                        /* Although this is not considered as a transient error by curl,
+                         * this error can happen momentarily while casync is retrieving
+                         * all the chunks from a remote. In this case we want to give
+                         * a break to the server and retry.
+                         */
+                        usleep(sleep_time * trial);
+                        trial++;
+                        break;
```

Comment:
mm, could you maybe add a log_info() here, saying that a retry is attempted?

Reply:
done